src/compiler/checker.ts

```diff
@@ -17495,16 +17500,31 @@ namespace ts {
                 }
                 else if (tryElaborateArrayLikeErrors(source, target, /*reportErrors*/ false)) {
                     if (props.length > 5) { // arbitrary cutoff for too-long list form
-                        reportError(Diagnostics.Type_0_is_missing_the_following_properties_from_type_1_Colon_2_and_3_more, typeToString(source), typeToString(target), map(props.slice(0, 4), p => symbolToString(p)).join(", "), props.length - 4);
+                        if (isJsxOpeningLikeElementAttributesError(errorNode)) {
+                            reportError(Diagnostics.This_0_tag_is_missing_the_following_attributes_Colon_1_and_2_more, getPropertiesPathName(errorNode as JsxTagNameExpression, true)?.join("."), map(props.slice(0, 4), p => symbolToString(p)).join(", "), props.length - 4);
+                        }
+                        else {
+                            reportError(Diagnostics.Type_0_is_missing_the_following_properties_from_type_1_Colon_2_and_3_more, typeToString(source), typeToString(target), map(props.slice(0, 4), p => symbolToString(p)).join(", "), props.length - 4);
+                        }
                     }
                     else {
-                        reportError(Diagnostics.Type_0_is_missing_the_following_properties_from_type_1_Colon_2, typeToString(source), typeToString(target), map(props, p => symbolToString(p)).join(", "));
+                        if (isJsxOpeningLikeElementAttributesError(errorNode)) {
+                            reportError(Diagnostics.This_0_tag_is_missing_the_following_attributes_Colon_1, getPropertiesPathName(errorNode as JsxTagNameExpression, true)?.join("."), map(props, p => symbolToString(p)).join(", "));
+                        }
+                        else {
+                            reportError(Diagnostics.Type_0_is_missing_the_following_properties_from_type_1_Colon_2, typeToString(source), typeToString(target), map(props, p => symbolToString(p)).join(", "));
+                        }
                     }
                     if (shouldSkipElaboration && errorInfo) {
                         overrideNextErrorInfo++;
                     }
                 }
                 // No array like or unmatched property error - just issue top level error (errorInfo = undefined)
+
+                function isJsxOpeningLikeElementAttributesError(node: Node | undefined) {

```

Comment:
Isn't there a way to pass a flag so that you don't have to check whether you're *probably* trying to check against the attributes?

Reply:
Not pretty sure whether I got the point, but I think there is no such flag existing, you mean passing in a new one?

Reply:
Yeah, adding a new parameter in the function that contains calls to `isJsxOpeningLikeElementAttributesError`